app/pages/SearchResultsPage/SearchResultsPage.tsx

```diff
@@ -25,24 +25,42 @@ export const SearchResultsPage = () => {
   const [lastPush, setLastPush] = useState(Date.now());
   const { search } = useLocation();
   const [expandList, setExpandList] = useState(false);
-  const searchState = useMemo(() => qs.parse(search.slice(1)), [search]);
+
+  // Todo: I'm not sure if this is the greatest solution; querying the searchState
+  // for the configure.aroundRadius property causes a TS type warning unless I
+  // define the interfaces below
+  interface ConfigureState {
+    aroundRadius?: string;
+    [key: string]: any;
+  }
+
+  interface SearchState {
+    configure?: ConfigureState;
+    [key: string]: any;
+  }
+
+  const searchState: SearchState = useMemo(() => qs.parse(search.slice(1)), [search]);
+  const [searchRadius, setSearchRadius] = useState(searchState?.configure?.aroundRadius || 'all');
```

Comment:
Ah nice. I hadn't known about that operator. Thanks!

That is a good catch. I added code to set any new text queries to the existing search state when a user enters a new query. I think it would probably be better to use one source of truth across components (perhaps using an appContext prop?) rather than doing this; I spent some time trying to rig that up but couldn't quite get it right without bugs. Thus, in the interest of time I went with this quicker solution.... I might be revisiting this again as part of my next chunk of work, as we will need to add another search box to the page per the designs. I could try to refactor it then. Not sure. Let me know if you have thoughts. Thank you

Reply:
Yeah, I think we can revisit it later. I think I had made an attempt at that in the past, and I can to the conclusion that we have to do something special on the initial page load, since we should first try to read from the URL, but after the initial page load, we should treat our internal app state as the source of truth. I think there was also some weird interaction with react-router as well, so we probably need to keep that in mind.